tests/kernel/common/src/irq_offload.c

```diff
@@ -35,10 +35,93 @@ static void offload_function(const void *param)
  *
  * @ingroup kernel_interrupt_tests
  *
- * @details Check whether offloaded running function is in interrupt
- * context, on the IRQ stack or not.
+ * @details
+ * Test Objective:
+ * - To verify the kernel architecture layer shall provide a means to raise an
+ *   interrupt from thread code that will run a user-provided C function under
+ *   the interrupt context.
+ * - We check whether offloaded running function is in interrupt context, on
+ *   the IRQ stack or not.
+ *
+ * Testing techniques:
+ * - Interface testing, function and block box testing,
+ *   dynamic analysis and testing
+ *
+ * Prerequisite Conditions:
+ * - CONFIG_IRQ_OFFLOAD=y was set.
+ *
+ * Input Specifications:
+ * - A pointer of C offload function
+ * - A pointer of passing parameter
+ *
+ * Test Procedure:
+ * -# Call irq_offload with parameter offload_function and SENTINEL_VALUE.
+ * -# In the C offload_function handler, call k_is_in_isr() to check if it is
+ *  in interrupt context currently.
+ * -# Store the passing argument into global variable sentinel.
+ * -# In main thread, check if the global variable sentinel equals to
+ *  SENTINEL_VALUE.
+ *
+ * Expected Test Result:
+ * - The running function is running on interrupt context.
+ *
+ * Pass/Fail Criteria:
+ * - Successful if check points in test procedure are all passed, otherwise
+ *   failure.
+ *
+ * Assumptions and Constraints:
+ * - N/A.
  */
 void test_irq_offload(void)
+{
+	/**TESTPOINT: Offload to IRQ context*/
+	irq_offload(offload_function, (const void *)SENTINEL_VALUE);
+
+	zassert_equal(sentinel, SENTINEL_VALUE,
+		      "irq_offload() didn't work properly");
+}
+
+/**
+ * @brief Verify nested irq lock
+ *
+ * @ingroup kernel_interrupt_tests
+ *
+ * @details
+ * Test Objective:
+ * - The kernel architecture layer shall provide a mechanism to restore the
+ *   local interrupt state saved prior to a mask operation.
+ * - We simply check lock function twice to make it is nested, then check if
+ *   unlock function works properly.
+ *
+ * Testing techniques:
+ * - Interface testing, function and block box testing,
+ *   dynamic analysis and testing
+ *
+ * Prerequisite Conditions:
+ * - N/A
+ *
+ * Input Specifications:
+ * - N/A
+ *
+ * Test Procedure:
+ * -# In main thread, call arch_irq_lock() and store it's return value in key1.
+ * -# Call arch_irq_unlocked(key1), and check if it return true.
+ * -# call arch_irq_lock() and store it's return value in key2.
+ * -# Call arch_irq_unlocked(key2), and check if it return false.
```

Comment:
at line 108, 109 and 110, "return" should be "returns".

Reply:
Got it fix, thanks. I'll apply this to all other PRs which are like this.